src/python/pants/option/config.py

```diff
@@ -192,14 +192,10 @@ def has_section(self, section):
     return self.configparser.has_section(section)
 
   def has_option(self, section, option):
-    return (self.configparser.has_option(section, option) or
-            self.configparser.has_option(self._DEFAULT_SECTION, option))
+    return self.configparser.has_option(section, option)
 
   def get_value(self, section, option):
-    if self.configparser.has_option(section, option):
-      return self.configparser.get(section, option)
-    else:
-      return self.configparser.get(self._DEFAULT_SECTION, option)
+    return self.configparser.get(section, option)
```

Comment:
I thought the ini parser did the fallback to the DEFAULT section itself. Was that not actually the case? Does it only do interpolation? In which case, isn't this a major change in how our config parser works?